### DISCUSSION TITLE
Reload a trial object from a study

### DISCUSSION THREAD

**Discussion by matthieucx** [optuna/optuna#4630] [Q&A]
Hello!

I am using the Ask and Tell interface to sample parameters and launch training dags in Airflow. I would like to incorporate pruning and would need to call trial.report().

Is there a way to load already created trials from a study ? I tried 
`trial study.trials[trial_id]` 
`study.tell(trial, value)` and it does not seem to work.

https://github.com/optuna/optuna/issues/2177
Issues and PR related here makes me think that nothing like that is present, is that so?

Thanks in advance!

**Comment by nzw0301** [comment]
If the trial has already been evaluated, we need to recreate a new trial from its param. 

```python
import optuna

def objective(t):
    x = t.suggest_int("x", -100, 100)
    y = t.suggest_int("y", 2, 10000, log=True)
    return x + y
        
original_study = optuna.create_study()
original_study.optimize(objective, n_trials=20)

another_study = optuna.create_study()

# ex. adding the last trial without its objective value in the `original_study` to the new one.
another_study.enqueue_trial(original_study.trials[-1].params)


def objective(t):
    x = t.suggest_int("x", -100, 100)
    y = t.suggest_int("y", 2, 10000, log=True)
    for step in range(100):
        t.report(value=x+y+step, step=step)
    return x + y

# evaluate the added trial from `original_study` and perform a single new trial.
original_study.optimize(objective, n_trials=2)
```

I believe that the basic idea works for the ask-and-tell interface too.

**Reply by matthieucx** [reply]
Thanks for your answer.

In my use case, Process A creates a bunch of trials and sample their parameters. Then, Process B fetch the list of created trials and launches a training job for each task. So trials are created and evaluated in two different processes. That's perfectly fine with Ask and Tell using the trial ID, but I need to reload a trial, not create a new one with the same parameters, to use trial.report().

Does your answer implies that there is a way to reload a trial if it hasn't been evaluated yet ?

**Reply by nzw0301** [reply]
Can you share the code?
